File: app/src/main/java/com/acrcloud/data/Status.java
package com.acrcloud.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Status implements Parcelable {

    @SerializedName("msg")
    @Expose
    private String msg;
    @SerializedName("code")
    @Expose
    private Integer code;
    @SerializedName("version")
    @Expose
    private String version;
    public final static Creator<Status> CREATOR = new Creator<Status>() {


        @SuppressWarnings({
                "unchecked"
        })
        public Status createFromParcel(Parcel in) {
            return new Status(in);
        }

        public Status[] newArray(int size) {
            return (new Status[size]);
        }

    };

    protected Status(Parcel in) {
        this.msg = ((String) in.readValue((String.class.getClassLoader())));
        this.code = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.version = ((String) in.readValue((String.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     */
    public Status() {
    }

    /**
     * @param code
     * @param msg
     * @param version
     */
    public Status(String msg, Integer code, String version) {
        super();
        this.msg = msg;
        this.code = code;
        this.version = version;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(msg);
        dest.writeValue(code);
        dest.writeValue(version);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/com/acrcloud/data/Status.kt
package com.acrcloud.data

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Status : Parcelable {

    @SerializedName("msg")
    @Expose
    var msg: String? = null
    @SerializedName("code")
    @Expose
    var code: Int? = null
    @SerializedName("version")
    @Expose
    var version: String? = null

    protected constructor(`in`: Parcel) {
        this.msg = `in`.readValue(String::class.java.classLoader) as String
        this.code = `in`.readValue(Int::class.java.classLoader) as Int
        this.version = `in`.readValue(String::class.java.classLoader) as String
    }

    /**
     * No args constructor for use in serialization
     */
    constructor() {}

    /**
     * @param code
     * @param msg
     * @param version
     */
    constructor(msg: String, code: Int?, version: String) : super() {
        this.msg = msg
        this.code = code
        this.version = version
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(msg)
        dest.writeValue(code)
        dest.writeValue(version)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Status> {
        override fun createFromParcel(parcel: Parcel): Status {
            return Status(parcel)
        }

        override fun newArray(size: Int): Array<Status?> {
            return arrayOfNulls(size)
        }
    }


}
